ui/index.js

```diff
@@ -122,6 +124,10 @@ async function startApp (metamaskState, backgroundConnection, opts) {
     },
   }
 
+  if (password) {
+    !metamaskState.isUnlocked && await store.dispatch(actions.tryUnlockMetamask(password))
```

Comment:
This would be much faster if moved into the background. This will delay the initialization of the UI the first time it loads, which is already pretty slow as-is